source/dem/force_chains_visualization.cc

```diff
@@ -0,0 +1,511 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2020 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ */
+
+#include <core/parameters_lagrangian.h>
+#include <core/tensors_and_points_dimension_manipulation.h>
+
+#include <dem/force_chains_visualization.h>
+#include <dem/particle_particle_contact_force.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+using namespace DEM;
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  ParticlesForceChains(const DEMSolverParameters<dim> &dem_parameters_in)
+  : ParticleParticleContactForce<dim, contact_model, rolling_friction_model>(
+      dem_parameters_in)
+  , dem_parameters(dem_parameters_in)
+{
+  force_normal.push_back(0);
+  vertices.push_back(Point<3>(0, 0, 0));
+  vertices.push_back(Point<3>(0, 0, 0));
+}
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+void
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  multi_general_cell(Triangulation<1, 3>         &tria,
+                     const std::vector<Point<3>> &vertices)
+{
+  const unsigned int       n_cells = vertices.size() / 2;
+  std::vector<CellData<1>> cells(n_cells, CellData<1>());
+  for (unsigned int i = 0; i < n_cells; ++i)
+    {
+      for (unsigned int j = 0; j < 2; ++j)
+        {
+          cells[i].vertices[j] = 2 * i + j;
+          cells[i].material_id = 0;
+        }
```

Comment:
since this loop is always 2, do we really need it? You could replace it with just the 3 lines.